### DISCUSSION TITLE
@preact/signals-react v2.0.0 breaking changes

### DISCUSSION THREAD

**Discussion by javierhersan** [preactjs/signals#539] [General]
Hello to everyone, could someone explain me why the following BREAKING CHANGES have been introduced in v2.0.0.

I will give an example. I have the following global state with signals.

![image](https://github.com/preactjs/signals/assets/37757627/7d5ff1ea-b75c-4fee-a95a-bfe7fc9e1e97)

And if i want my component to react to a signal change now I have to use useSignals(), whereas before this was not necessary.

![image](https://github.com/preactjs/signals/assets/37757627/6b2c28b2-e11d-4d8b-af5a-c600f38ec9d0)

I prefer v1, I do not want to use useSignals() and neither Babel is too slow and outdated.

Why was this BREAKING CHANGE introduced?

**Comment by rschristian** [comment]
Because there was a number of edge cases w/ v1 that could not ever be resolved (not to mention the React team was very vocal about their distaste for it).

As React doesn't offer a plugin API, and shows absolutely no interest in offering one, this is the next-best option.

> Babel is too slow and outdated.

Babel isn't slow, but, if you did have issues (and I urge you to benchmark first), you could write an SWC/etc port instead.

**Reply by javierhersan** [reply]
Hi @rschristian, thanks for your reply, sorry if I sound rude, I've read again my message, and it was not my intention.

Can we include something to maintain backwards compatibility with v1, or will the bundle size increase too much? I like the simplicity it had before.

**Reply by rschristian** [reply]
I'd recommend trying out Preact in that case as no, this is the direction of the React integration.

You can still access it with `@preact/signals-react/auto`, but again, this behavior has bugs that can't be fixed and it will be removed at some point in the future.
